atlasdb-config/src/test/java/com/palantir/atlasdb/http/AtlasDbHttpClientsTest.java

```diff
@@ -99,17 +112,33 @@ public void setup() {
                 getUriForPort(availablePort));
     }
 
+    @Test
+    public void payloadLimitingClientThrowsOnRequestThatIsTooLarge() {
+        TestResource client = AtlasDbHttpClients.createProxy(new MetricRegistry(), NO_SSL, getUriForPort(availablePort),
+                TestResource.class, true);
+        assertThat(client.postRequest(new byte[AtlasDbInterceptors.MAX_PAYLOAD_SIZE / 2])).isTrue();
+        assertThatThrownBy(() -> client.postRequest(new byte[AtlasDbInterceptors.MAX_PAYLOAD_SIZE]))
+            .isInstanceOf(IllegalArgumentException.class);
```

Comment:
either throw a more specific exception, or assert what the message may contain (does not have to be exact).

Reply:
it would be nice if in line 119, you have something like:

```java
assertThat(client.postRequest(new byte[AtlasDbInterceptors.MAX_PAYLOAD_SIZE / 2]))
  .as("payload with size under limit is fine")
  .isTrue();
```
and 
```java
assertThatThrownBy(() -> client.postRequest(new byte[AtlasDbInterceptors.MAX_PAYLOAD_SIZE]))
  .as("payload with size at the limit throws") 
  .isInstanceOf(IllegalArgumentException.class)
  .withMessageContaining("Bla bla bla");
```

I've typically found adding `as` descriptions pretty nice if you're trying to directly distinguish two bits of behaviour

Reply:
👍